src/Sarif/NotYetGenerated/SarifRewritingVisitor.cs

```diff
@@ -684,7 +726,14 @@ public virtual Run VisitRun(Run node)
                         var value = node.Files[key];
                         if (value != null)
                         {
-                            node.Files[key] = VisitNullChecked(value);
+                            string newKey = key;
+                            node.Files.Remove(key);
+                            value = VisitNullChecked(value, ref newKey);
+
+                            if (newKey != null)
+                            {
+                                node.Files[newKey] = VisitNullChecked(value);
```

Comment:
>VisitNullChecked [](start = 53, length = 16)

Doesn't this cause us to visit the FileData/FileLocation twice. Once from visiting the dictionaryentry and again here?